chalice/app.py

```diff
@@ -121,7 +138,8 @@ def wrapper(*args, **kwargs):
                     chalice_response = future.result(timeout=time_remaining_s)
                     return chalice_response
                 except TimeoutError:
-                    return timeout_response()
+                    request_method, request_path = determine_path(method)
+                    return timeout_response(method=request_method, uri=request_path)
```

Comment:
Consider moving the logic of `determine_path` into `timeout_response`, since that is the only place it's used.

Reply:
Will do.